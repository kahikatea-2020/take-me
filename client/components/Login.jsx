import React from 'react'
import { Link } from 'react-router-dom'
import { Form } from 'semantic-ui-react'
import { connect } from 'react-redux'

import { isAuthenticated, signIn } from 'authenticare/client'

class Login extends React.Component {
  state = {
    username: '',
    password: ''
  }

  updateField = e => {
    this.setState({
      [e.target.name]: e.target.value
    })
  }

  submitHandler = e => {
    signIn({
      username: this.state.username,
      password: this.state.password
    }, {
      baseUrl: process.env.BASE_API_URL
    })
      .then((token) => {
        if (isAuthenticated()) {
          this.props.history.push('/')
        }
      })
  }

  render () {
    return (
      <>
        <h1>Login</h1>
        <Form>
          <Form.Input
            onKeyUp={this.updateField}
            fluid
            width={6}
            name='username'
            placeholder='Username'
            type='text'
          />
          <Form.Input
            onKeyUp={this.updateField}
            fluid
            width={6}
            name='password'
            placeholder='Password'
            type='password'
            autoComplete='off'
          />
          <Form.Group>
            <Link to='/'>
              <Form.Button>
              Cancel
              </Form.Button>
            </Link>
            <Form.Button
              type='submit'
              onClick={this.submitHandler}
            >
              Submit
            </Form.Button>
          </Form.Group>
        </Form>
      </>
    )
  }
}

const mapStateToProps = state => {
  return {
    state
  }
}

export default connect(mapStateToProps)(Login)
