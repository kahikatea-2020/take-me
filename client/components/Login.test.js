import React from 'react'
import renderer from 'react-test-renderer'

import Login from './Login'

test('Test Login component so it renders right', () => {
  const rendered = renderer
    .create(<Login />)
    .toJSON()
  expect(rendered).toMatchSnapshot()
})
