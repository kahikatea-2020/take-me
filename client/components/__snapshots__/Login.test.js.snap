// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Login component so it renders right 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="wrapper"
      >
        <div
          id="login"
        >
          <h1>
            Login
          </h1>
          <form
            class="ui form"
            id="login-cont"
          >
            <div
              class="fields"
            >
              <div
                class="six wide field"
              >
                <div
                  class="ui fluid input"
                >
                  <input
                    name="username"
                    placeholder="Username"
                    type="text"
                  />
                </div>
              </div>
            </div>
            <div
              class="fields"
            >
              <div
                class="six wide field"
              >
                <div
                  class="ui fluid input"
                >
                  <input
                    autocomplete="off"
                    name="password"
                    placeholder="Password"
                    type="password"
                  />
                </div>
              </div>
            </div>
            <div
              class="fields"
              id="buttons"
            >
              <a
                href="/"
              >
                <div
                  class="field"
                >
                  <button
                    class="ui button"
                  >
                    Cancel
                  </button>
                </div>
              </a>
              <div
                class="field"
              >
                <button
                  class="ui button"
                  type="submit"
                >
                  Submit
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      id="wrapper"
    >
      <div
        id="login"
      >
        <h1>
          Login
        </h1>
        <form
          class="ui form"
          id="login-cont"
        >
          <div
            class="fields"
          >
            <div
              class="six wide field"
            >
              <div
                class="ui fluid input"
              >
                <input
                  name="username"
                  placeholder="Username"
                  type="text"
                />
              </div>
            </div>
          </div>
          <div
            class="fields"
          >
            <div
              class="six wide field"
            >
              <div
                class="ui fluid input"
              >
                <input
                  autocomplete="off"
                  name="password"
                  placeholder="Password"
                  type="password"
                />
              </div>
            </div>
          </div>
          <div
            class="fields"
            id="buttons"
          >
            <a
              href="/"
            >
              <div
                class="field"
              >
                <button
                  class="ui button"
                >
                  Cancel
                </button>
              </div>
            </a>
            <div
              class="field"
            >
              <button
                class="ui button"
                type="submit"
              >
                Submit
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
    Symbol(observable): [Function],
  },
  "unmount": [Function],
}
`;
